ecommerce/src/components/basket_page/DrawProductCardFromTheBasket.tsx

```diff
@@ -0,0 +1,179 @@
+import { LineItem } from '@commercetools/platform-sdk';
+import { Button } from 'antd';
+import { useContext, useEffect, useState } from 'react';
+import { JsxElement } from 'typescript';
+import { Context } from '../..';
+import {
+    addProductItemAnonim,
+    addProductItemCastomer,
+    changeProductAnonim,
+    changeProductAuth,
+    createAnonimusCart,
+    createCartAuth,
+    getCartsAnonimus,
+    getCartsAuth,
+    getQuantityAnonimus,
+    getQuantityAuth,
+    removeItemAnonim,
+    removeItemCastomer,
+    removeProductItemAnonim,
+    removeProductItemCastomer,
+} from '../catalog_page/requests';
+import { getLocalStorage } from '../login_page/BuildClient';
+import { CartType } from './BasketStore';
+import classes from './DrawProductCard.module.css';
+
+type PropsInterface = {
+    product: LineItem;
+};
+
+/* const addProduct = () => {
+    console.log('add product');
+}; */
+
+/* const deleteProduct = () => {
+    console.log('product less');
+}; */
+
+const DrawProductCardFromTheBasket: React.FC<{ product: LineItem }> = (props: PropsInterface) => {
+    const { store, cart } = useContext(Context);
+    const [quantity, setQuantity] = useState(props.product.quantity);
+
+    // name of flowers
+    const namePlants = props.product.name.ru;
+
+    // image
+    const imagePlantsArr = props.product.variant.images;
+    const imageProduct: string = imagePlantsArr ? imagePlantsArr[0].url : './defaultImage.jpg';
+
+    // price of product
+    const priceProduct = props.product.price.discounted?.value.centAmount;
+
+    // count in basket
+    const countProduct1 = props.product.quantity;
+
+    // promoCode
+
+    let priceAfterPromoCode: number | undefined;
+
+    const arrTemp = props.product.discountedPricePerQuantity;
+    if (arrTemp.length > 0) {
+        priceAfterPromoCode = arrTemp[0].discountedPrice.value.centAmount;
+    }
+
+    // count in basket
+    // let countProduct = props.product.quantity;
+    let countProduct = quantity;
+    const handleEvent = async () => {
+        if (store.isAuth) {
+            const tokenStore = getLocalStorage();
+            const { refreshToken } = tokenStore;
+            // const getCartsCastomer = await getCartsAuth(refreshToken);
```

Comment:
it would be better to add separate file for auth and move there all logic related with authorization